great_expectations/datasource/fluent/sources.py

```diff
@@ -512,21 +517,14 @@ def update_datasource(
                 self._data_context.datasources.get(datasource_name), "id", None
             )
             if id_:
-                # if not a str `name_or_datasource` is a datasource and `id` can be directly attached
-                if name_or_datasource and not isinstance(name_or_datasource, str):
-                    name_or_datasource.id = id_
-                else:
-                    kwargs["id"] = id_
+                updated_datasource.id = id_
 
-            # local delete only, don't update the persisted store entry
-            self._data_context._delete_fluent_datasource(
-                datasource_name=datasource_name, _call_store=False
-            )
-            # Now that the input is validated and the old datasource is deleted we pass the
-            # original arguments to the add method (ie name and not datasource_name).
-            return self.create_add_crud_method(datasource_type)(
-                name_or_datasource, **kwargs
+            updated_datasource.test_connection()
```

Comment:
@Kilo59 @cdkini - during our pairing session, @Kilo59 mentioned that `test_connection` should only be called during `add_datasource` operation, not `update_datasource`. However, when I went back to double check, it looks like adding `.test_connection()` (in `update_datasource` and `add_or_update_datasource`) is actually required to maintain existing behavior, since existing behavior is to delete then re-add Datasources when doing updates. Since re-adding just uses existing `add_datasource` method, `test_connection()` ends up getting called, just not in the main body of the update methods.

Should we maintain this behavior or is the existing impl a bug? LMK if I'm misreading the logic somewhere.

Reply:
That makes, sense.
I wouldn't have expected that but given that's our current behavior, I say let's keep it.